pkg/audit/manager.go

```diff
@@ -233,58 +246,70 @@ func (am *Manager) auditResources(ctx context.Context) ([]*constraintTypes.Resul
 		}
 	}
 
-	var responses []*constraintTypes.Result
 	var errs opa.Errors
 	nsCache := newNSCache()
 
 	for gv, gvKinds := range clusterAPIResources {
 		for kind := range gvKinds {
 			objList := &unstructured.UnstructuredList{}
-			objList.SetGroupVersionKind(schema.GroupVersionKind{
-				Group:   gv.Group,
-				Version: gv.Version,
-				Kind:    kind + "List",
-			})
-
-			err := am.client.List(ctx, objList)
-			if err != nil {
-				am.log.Error(err, "Unable to list objects for gvk", "group", gv.Group, "version", gv.Version, "kind", kind)
-				continue
+			opts := &client.ListOptions{
+				Limit: int64(*auditListLimit),
 			}
 
-			for _, obj := range objList.Items {
-				if am.skipExcludedNamespace(obj.GetNamespace()) {
+			for {
+				objList.SetGroupVersionKind(schema.GroupVersionKind{
+					Group:   gv.Group,
+					Version: gv.Version,
+					Kind:    kind + "List",
+				})
+
+				err := am.client.List(ctx, objList, opts)
+				if err != nil {
+					am.log.Error(err, "Unable to list objects for gvk", "group", gv.Group, "version", gv.Version, "kind", kind)
 					continue
 				}
 
-				ns := corev1.Namespace{}
-				if obj.GetNamespace() != "" {
-					ns, err = nsCache.Get(ctx, am.client, obj.GetNamespace())
-					if err != nil {
-						am.log.Error(err, "Unable to look up object namespace", "group", gv.Group, "version", gv.Version, "kind", kind)
+				for _, obj := range objList.Items {
+					if am.skipExcludedNamespace(obj.GetNamespace()) {
 						continue
 					}
-				}
 
-				augmentedObj := target.AugmentedUnstructured{
-					Object:    obj,
-					Namespace: &ns,
+					ns := corev1.Namespace{}
+					if obj.GetNamespace() != "" {
+						ns, err = nsCache.Get(ctx, am.client, obj.GetNamespace())
+						if err != nil {
+							am.log.Error(err, "Unable to look up object namespace", "group", gv.Group, "version", gv.Version, "kind", kind)
+							continue
+						}
+					}
+
+					augmentedObj := target.AugmentedUnstructured{
+						Object:    obj,
+						Namespace: &ns,
+					}
+					resp, err := am.opa.Review(ctx, augmentedObj)
+					if err != nil {
+						errs = append(errs, err)
+					} else if len(resp.Results()) > 0 {
+						err = am.getUpdateListsFromAuditResponses(updateLists, resp.Results(), totalViolationsPerConstraint, totalViolationsPerEnforcementAction)
```

Comment:
This function should be renamed something like `addAuditResponsesToUpdateLists` to reflect that we are no longer getting the list.

Reply:
updated